pkg/datapath/linux/ipsec.go

```diff
@@ -258,6 +262,36 @@ func (n *linuxNodeHandler) enableIPsecIPv4(newNode *nodeTypes.Node, nodeID uint1
 			if err != nil {
 				statesUpdated = false
 			}
+
+			// In Encrypt Overlay mode, outermost header is ESP tunnel.
+			// Packet format : [IP|ESP|IP|VxLAN|<payload>]
+			// ESP tunnel src/dst addresses are underlay IPs of the node (NodeInternalIP).
+			// VxLAN tunnel src/dst addresses are also underlay IPs of the node (NodeInternalIP).
+			if n.nodeConfig.EnableIPSecEncryptedOverlay {
+				localUnderlayIP := n.nodeAddressing.IPv4().PrimaryExternal()
+				if localUnderlayIP == nil {
+					return false, errs
+				}
+				remoteUnderlayIP := newNode.GetNodeIP(false)
+				if remoteUnderlayIP == nil {
+					return false, errs
+				}
```

Comment:
Instead of retrieving this here a second time (and in a different way), how about we move `localNodeInternalIP`/`remoteNodeInternalIP` above (lines 205--210) outside of the if/else and use them in both the `subnetEncryption` case and this `encryptedOverlay` case?

Reply:
I did more digging, `n.getV4LinkLocalIP()` does not return Node Internal IP. In case of tunnel configuration, it picks tunnel interface. 

https://github.com/cilium/cilium/blob/main/pkg/datapath/linux/ipsec.go#L30-L55

Reply:
Isn't that a bug in `getLinkLocalIP`? cilium_vxlan doesn't have an IP address.

Reply:
Hi @pchaigno, any concerns in resolving this comment ?